packages/widget/src/controllers/transfers/fungible-token-transfer.ts

```diff
@@ -133,9 +167,23 @@ export class FungibleTokenTransferController implements ReactiveController {
       return;
     }
     this.sourceNetwork = network;
+    if (this.sourceNetwork.type === Network.SUBSTRATE) {
+      this.setSubstrateProvider(this.sourceNetwork);
+    }
     void this.filterDestinationNetworksAndResources(network);
   };
 
+  setSubstrateProvider = (sourceNetwork: Domain | undefined): void => {
+    const providers = this.substrateProviderContext.value?.substrateProviders;
+    if (!providers || !sourceNetwork?.chainId) {
+      return;
+    }
+
+    const provider = providers.get(sourceNetwork?.chainId);
```

Comment:
```suggestion
    const provider = providers.get(sourceNetwork.chainId);
```

Reply:
Can you open issue for filtering supported source networks based on substrate providers we received from user?